thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -702,11 +709,51 @@ class ThinReplicaImpl {
       LOG_ERROR(logger_, "Failed to read from root cert - popen() failed, error: " << strerror(errno));
       throw std::runtime_error("Failed to read from root cert - popen() failed!");
     }
+
+    delim = (use_unified_certs) ? "O = " : "OU = ";
+
     while (fgets(buffer.data(), buffer.size(), pipe_ptr.get()) != nullptr) {
       result = buffer.data();
       // parse the OU i.e., the client id from the subject field
-      cert_ou_field_set.insert(parseClientIdFromSubject(result));
+      cert_ou_field_set.insert(parseClientIdFromSubject(result, delim));
+    }
+  }
+
+  static void getClientIdSetFromRootCert(logging::Logger logger_,
+                                         const std::string& root_cert_path,
+                                         std::unordered_set<std::string>& cert_ou_field_set) {
+    for (auto& p : fs::recursive_directory_iterator(root_cert_path)) {
+      if ((p.path().filename().string()).compare("node.cert") == 0) {
+        getClientIdFromRootCert(logger_, p.path().string(), cert_ou_field_set, true);
+      }
+    }
+  }
+
+  static std::string getClientIdFromCertificate(const std::string& client_cert_path, bool use_unified_certs = false) {
+    std::array<char, 128> buffer;
```

Comment:
Don't use a hard-coded value. Define a constant data member for it. Not sure that 128 is a good value choice. Where does it come from?

Reply:
I am not sure, but was used earlier the same. I have added certSubjectLength = 555. 555 is the max subject length a certificate has.